packages/firestore/src/local/indexeddb_remote_document_cache.ts

```diff
@@ -95,66 +96,62 @@ export class IndexedDbRemoteDocumentCache implements RemoteDocumentCache {
+   * Adds the supplied entries to the cache.
+   *
+   * Callees must ensure that the total document size in the cache is updated
+   * after all entries are written (via `updateMetadata()`).
    */
-  addEntries(
+  addEntry(
     transaction: PersistenceTransaction,
-    entries: Array<{ key: DocumentKey; doc: DbRemoteDocument }>,
-    sizeDelta: number
+    key: DocumentKey,
+    doc: DbRemoteDocument
   ): PersistencePromise<void> {
-    const promises: Array<PersistencePromise<void>> = [];
-
-    if (entries.length > 0) {
-      const documentStore = remoteDocumentsStore(transaction);
-      let changedKeys = documentKeySet();
-      for (const { key, doc } of entries) {
-        promises.push(documentStore.put(dbKey(key), doc));
-        changedKeys = changedKeys.add(key);
-
-        promises.push(
-          this.indexManager.addToCollectionParentIndex(
-            transaction,
-            key.path.popLast()
-          )
-        );
-      }
-
-      if (this.keepDocumentChangeLog) {
-        promises.push(
-          documentChangesStore(transaction).put({
-            changes: this.serializer.toDbResourcePaths(changedKeys)
-          })
-        );
-      }
-
-      promises.push(this.updateSize(transaction, sizeDelta));
-    }
-
-    return PersistencePromise.waitFor(promises);
+    const documentStore = remoteDocumentsStore(transaction);
+    return documentStore.put(dbKey(key), doc).next(() => {
+      this.indexManager.addToCollectionParentIndex(
+        transaction,
+        key.path.popLast()
+      );
+    });
   }
 
   /**
-   * Removes a document from the cache. Note that this method does *not* do any
-   * size accounting. It is the responsibility of the caller to count the bytes removed
-   * and issue a final updateSize() call after removing documents.
+   * Removes a document from the cache.
+   *
+   * Callees must ensure that the total document size in the cache is updated
+   * after all entries are written (via `updateMetadata()`).
    *
    * @param documentKey The key of the document to remove
    * @return The size of the document that was removed.
    */
   removeEntry(
     transaction: PersistenceTransaction,
     documentKey: DocumentKey
-  ): PersistencePromise<number> {
-    // We don't need to keep changelog for these removals since `removeEntry` is
-    // only used for garbage collection.
+  ): PersistencePromise<void> {
     const store = remoteDocumentsStore(transaction);
     const key = dbKey(documentKey);
-    return store.get(key).next(document => {
-      if (document) {
-        return store.delete(key).next(() => dbDocumentSize(document));
-      } else {
-        return PersistencePromise.resolve(0);
-      }
+    return store.delete(key);
+  }
+
+  /**
+   * Udpates the document change log and adds the given delta to the cached current size.
+   * Callers to `addEntry()` and `removeEntry()` *must* call this afterwards to update the
+   * cache's metadata.
```

Comment:
Again, we could potentially make this private and not worry about callers messing up.

Reply:
I reworded this to point users to the new API.